internal/cbohelper/cbohelper.go

```diff
@@ -0,0 +1,175 @@
+	return getUrlFromIP(ironicIP), nil
+}
+
+func (ch *CBOHelper) GenerateIronicConfig() (config ignition_config_types_32.Config, err error) {
+	ironicBaseURL, err := ch.getIronicServiceURL()
+	if err != nil {
+		return config, err
+	}
+	config.Ignition.Version = "3.2.0"
+	// TODO: this should probably get the proxy settings as well
+	ib, err := iccignition.New([]byte{}, []byte{}, ironicBaseURL, ch.config.BaremetalIronicAgentImage, "", "", "", "", "", "", "")
+	if err != nil {
+		return config, err
+	}
+	config.Storage.Files = []ignition_config_types_32.File{ib.IronicAgentConf()}
+	// TODO: sort out the flags (authfile...) and copy network
+	config.Systemd.Units = []ignition_config_types_32.Unit{ib.IronicAgentService(false)}
+	return config, err
+}
+
+func (ch *CBOHelper) readProvisioningCR() (*metal3iov1alpha1.Provisioning, error) {
+	// Fetch the Provisioning instance
+	instance := &metal3iov1alpha1.Provisioning{}
+	namespacedName := types.NamespacedName{Name: metal3iov1alpha1.ProvisioningSingletonName, Namespace: ""}
+	if err := ch.Client.Get(context.TODO(), namespacedName, instance); err != nil {
+		return nil, errors.Wrap(err, "unable to read Provisioning CR")
+	}
+	return instance, nil
+}
+func (ch *CBOHelper) getIronicIP(info *metal3iov1alpha1.Provisioning) (string, error) {
+	config := info.Spec
+	if config.ProvisioningNetwork != metal3iov1alpha1.ProvisioningNetworkDisabled && !config.VirtualMediaViaExternalNetwork {
+		return config.ProvisioningIP, nil
+	}
+	return ch.getPodHostIP(info.Namespace)
+}
+
+func getUrlFromIP(ipAddr string) string {
+	if strings.Contains(ipAddr, ":") {
+		// This is an IPv6 addr
+		return "https://" + fmt.Sprintf("[%s]", ipAddr)
+	}
+	if ipAddr != "" {
+		// This is an IPv4 addr
+		return "https://" + ipAddr
+	} else {
+		return ""
+	}
+}
+
+func (ch *CBOHelper) getPodHostIP(targetNamespace string) (string, error) {
+	metal3AppName := "metal3"
+	stateService := "metal3-state"
+	cboLabelName := "baremetal.openshift.io/cluster-baremetal-operator"
+
+	labelSelector := &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"k8s-app":    metal3AppName,
+			cboLabelName: stateService,
+		}}
+
+	selector, err := metav1.LabelSelectorAsSelector(labelSelector)
+	if err != nil {
+		return "", err
+	}
+
+	listOptions := client.ListOptions{LabelSelector: selector,
+		Namespace: targetNamespace,
+	}
+	podList := &corev1.PodList{}
+	err = ch.Client.List(context.Background(), podList, &listOptions)
+	if err != nil {
+		return "", err
+	}
+	var hostIP string
+	switch len(podList.Items) {
+	case 0:
+		err = fmt.Errorf("ironic IP not available yet")
+	case 1:
+		hostIP = podList.Items[0].Status.HostIP
+	default:
+		// We expect only one pod with the above LabelSelector
+		err = fmt.Errorf("there should be only one pod listed for the given label")
+	}
+	return hostIP, err
+}
+
+func (ch *CBOHelper) ConvergedFlowAvailable() bool {
+	return ch.convergedFlowAvailable
+}
+
+func (ch *CBOHelper) setConvergedFlowAvailability() {
```

Comment:
if i understand right we can't have hybrid option right? I mean we will run only in converged flow or not. Right? i can't remove label from infraenv to allow using not converged flow for example?

Reply:
I think will be nice to have such an option though

Reply:
We have an hybrid option (per infra env).
This is about the availability of the converged flow (4.11 and above)